.github/workflows/firesim-run-tests.yml

```diff
@@ -269,3 +269,48 @@ jobs:
           python3 -m sphinx.ext.intersphinx docs/_build/html/objects.inv
           cat /tmp/sphinx-err*.log
       - uses: ./.github/actions/job-end
+
+  clang-format:
+    name: clang-format
+    runs-on: ubuntu-20.04
+    steps:
+      # Clone the repository (shallow to save time).
+      - name: Checkout sources
+        uses: actions/checkout@v2
+        with:
+          fetch-depth: 2
+          submodules: false
+
+      # Install clang-format
+      - name: Install clang-format
+        run: |
+          sudo apt update
+          sudo apt install clang-format -y
+
+      # Run 'clang-format', comparing against the base commit hash. If anything
+      # got reformatted, fail and output a patch. This includes choosing the git
+      # commit to diff against for the purposes of linting. Since this workflow
+      # is triggered on both pushes and pull requests, we have to determine if
+      # the pull request target branch is set (which it will only be on the PR
+      # triggered flow). If it's not, then compare against the last commit.
+      # Since we did a shallow fetch for this repo, we must fetch the commit
+      # upon which we be diffing
+      - name: Run clang-format
+        run: |
+          if [ -z "${{ github.base_ref }}" ]; then
+            DIFF_COMMIT=$(git rev-parse HEAD^)
+          else
```

Comment:
So i'm a little confused here -- this workflow should only be run on open, non-draft PRs against main (and 1.[0-9].x), why _isn't_ the `base_ref` always set?

Reply:
I've removed this check. The rule was adapted from another CI job which did not run only on pull events that expose base_ref. However, if you think the triggers of this job will change, we should revert.